### PR TITLE
fix: thorchain & osmo loading state fixes

### PR DESCRIPTION
**Pull request by elmutt**
## Description

- isSubmitting was `true` for long periods of time if the user submitted a thor or osmo tx and then goes back to try a new transaction. this prevents the user from creating a new transaction because the button is `loading`. 

- To fix this we define our own isSubmittingLocal for this onSubmit

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

Low risk. Fixes a bug

## Testing

1) enable thorchain feature flag
2) Do a thorchain trade -- BCH -> ATOM is a cheap one for example
3) When the trade is pending click back on the trade widget and see that it is possible to re-enter new trade information without having to wait for the previous trade to complete

### Engineering

see testing section & screenshots

### Operations

see testing section & screenshots

## Screenshots (if applicable)

Previous behavior -- incorrectly spins until the previous trade completes and blocking user from making a new trade:
Screen Shot 2022-08-26 at 12 16 26 PM

### CODE REVIEW

File: src/components/Trade/TradeInput.tsx [hunk 1]
```diff
@@ -491,7 +491,7 @@ export const TradeInput = ({ history }: RouterProps) => {
                   ? 'red'
                   : 'blue'
               }
-              isLoading={isSubmitting || isSendMaxLoading}
```
Comment (gomesalexandre) [inline src/components/Trade/TradeInput.tsx:494]: I had initial concerns about this but it makes sense... kind of. `isSubmitting` is triggered on async form submit operations, and there's no way to programmatically set it back to false.

So while it does make sense not to react on this field for loading states, the implication of this is that there's now a user-noticeable delay between clicking "Preview" and being routed to the next route, which is not *not* instant.

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx [hunk 2]
```diff
@@ -146,13 +146,13 @@ export const TradeConfirm = ({ history }: RouterProps) => {
...
       case 'Osmosis':
-        return `${osmosisAsset?.explorerTxLink}${buyTxid}`
+        return `${osmosisAsset?.explorerTxLink}${sellTxid}`
       case 'CowSwap':
         return `https://explorer.cow.fi/orders/${sellTxid}`
       default:
         return `${trade.sellAsset?.explorerTxLink}${sellTxid}`
     }
-  }, [trade, sellTxid, osmosisAsset, buyTxid])
+  }, [trade, sellTxid, osmosisAsset])
```
Comment (gomesalexandre) [inline src/components/Trade/TradeConfirm/TradeConfirm.tsx:155]: Revert and make it a separate PR?

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
See comment, while this does fix the issue, we should revamp the loading state to accommodate for it, so that loading state still works as before between trade input and preview routes.

Additionally, I wasn't able to test OSMO<->ATOM swaps (both directions) because of Figment API 501s so I can't test the Tx explorer link fixes, but let's make it its separate PR to keep things atomic and ensure we can test these two changes separately.

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx (hunk 2 above)
Comment (elmutt) [reply src/components/Trade/TradeConfirm/TradeConfirm.tsx:155]: Why? That just slows things down

File: src/components/Trade/TradeInput.tsx (hunk 1 above)
Comment (elmutt) [reply src/components/Trade/TradeInput.tsx:494]: I havent seen a delay

File: src/components/Trade/TradeConfirm/TradeConfirm.tsx (hunk 2 above)
Comment (toshiSat) [reply src/components/Trade/TradeConfirm/TradeConfirm.tsx:155]: Are you really asking to split up a PR that is adding and removing 4 lines of code 😂

Comment (elmutt) [reply src/components/Trade/TradeConfirm/TradeConfirm.tsx:155]: https://github.com/shapeshift/web/pull/2576

File: src/components/Trade/TradeInput.tsx (hunk 1 above)
Comment (elmutt) [reply src/components/Trade/TradeInput.tsx:494]: Added state to track this onSubmit button only

File: src/components/Trade/TradeInput.tsx [hunk 3]
```diff
@@ -228,6 +230,7 @@ export const TradeInput = ({ history }: RouterProps) => {
     } catch (e) {
       showErrorToast(e)
     }
+    setIsUpdatingTrade(false)
   }
```
Comment (gomesalexandre) [inline src/components/Trade/TradeInput.tsx:234]:
```suggestion
    } finally {
      setIsUpdatingTrade(false)
    }
  }
```

**Review by gomesalexandre — APPROVED** [review]
Conceptually agree with this one and does what it says on the box for the purpose of fixing this.

Note that this surfaces two new issues:

- Submitting the exact same trade will result in `Error [Kind: Error] {"message":"failed to broadcast transaction: codespace: sdk, code: 19, description: tx already in mempool"}`
- Submitting a different trade will result in `Error [Kind: Error] {"message":"failed to broadcast transaction: codespace: sdk, code: 32, description: incorrect account sequence"}` - that's an error we've already seen with Cosmos staking, happening when trying to re-broadcast a un/de/relegate / claim Tx without first reloading the page

Are these issues we're aware of, or should we add them to the swapper dashboard?

**Comment by elmutt** [comment]
> Conceptually agree with this one and does what it says on the box for the purpose of fixing this.
> 
> Note that this surfaces two new issues:
> 
> * Submitting the exact same trade will result in `Error [Kind: Error] {"message":"failed to broadcast transaction: codespace: sdk, code: 19, description: tx already in mempool"}`
> * Submitting a different trade will result in `Error [Kind: Error] {"message":"failed to broadcast transaction: codespace: sdk, code: 32, description: incorrect account sequence"}` - that's an error we've already seen with Cosmos staking, happening when trying to re-broadcast a un/de/relegate / claim Tx without first reloading the page
> 
> Are these issues we're aware of, or should we add them to the swapper dashboard?

These should happen a lot less now with the latest swapper package. Can you consistently reproduce?
